presto-native-sidecar-plugin/src/main/java/com/facebook/presto/sidecar/sessionpropertyproviders/NativeSystemSessionPropertyProvider.java

```diff
@@ -50,28 +62,41 @@ public class NativeSystemSessionPropertyProvider
     private final NodeManager nodeManager;
     private final TypeManager typeManager;
     private final JsonCodec<List<SessionPropertyMetadata>> nativeSessionPropertiesJsonCodec;
+    private final Supplier<List<PropertyMetadata<?>>> memoizedSessionPropertiesSupplier;
+    private final HttpClient httpClient;
+    private static final String SESSION_PROPERTIES_ENDPOINT = "/v1/properties/session";
 
     @Inject
     public NativeSystemSessionPropertyProvider(
+            @ForSidecarInfo HttpClient httpClient,
             JsonCodec<List<SessionPropertyMetadata>> nativeSessionPropertiesJsonCodec,
             NodeManager nodeManager,
             TypeManager typeManager)
     {
         this.nativeSessionPropertiesJsonCodec = requireNonNull(nativeSessionPropertiesJsonCodec, "nativeSessionPropertiesJsonCodec is null");
         this.nodeManager = requireNonNull(nodeManager, "nodeManager is null");
         this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.httpClient = requireNonNull(httpClient, "typeManager is null");
+        this.memoizedSessionPropertiesSupplier = Suppliers.memoizeWithExpiration(this::fetchSessionProperties, 1, TimeUnit.HOURS);
```

Comment:
should we make this time limit configurable?

Reply:
Yes, let me add a config property for this.